### PR TITLE
[fix] faucet new rate limiter

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

Previous rate limiter is likely broken. This uses a better one with a per ip address strategy.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-faucet/src/server.rs [hunk 1]
```diff
@@ -53,6 +54,24 @@ pub async fn start_faucet(
...
+            .per_second(3600)
+            .burst_size(max_request_per_second as u32)
+            .finish()
+            .unwrap(),
+    );
+
+    let governor_limiter = governor_conf.limiter().clone();
+    let interval = Duration::from_secs(60);
+    // a separate background task to clean up
+    std::thread::spawn(move || loop {
```
Comment (amnn) [inline crates/sui-faucet/src/server.rs:69]: You're mixing thread-based scheduling (`thread::spawn`) and `tokio` based scheduling (`tokio::time::sleep`) here.  Stick to tokio. Look at the WAL cleanup task below for reference.

**Review by amnn — COMMENTED** [review]
Thanks @stefan-mysten, note that we don't need to enforce per-IP rate limits at the faucet level, because there is a load balancer in front of it that already does that, but we do need to fix the previously broken middleware.

File: crates/sui-faucet/src/server.rs (hunk 1 above)
Comment (stefan-mysten) [reply crates/sui-faucet/src/server.rs:69]: Ah yes, sorry. It was late and only fixed it in one place (the sleep one) :D

**Comment by stefan-mysten** [comment]
> Thanks @stefan-mysten, note that we don't need to enforce per-IP rate limits at the faucet level, because there is a load balancer in front of it that already does that, but we do need to fix the previously broken middleware.

Thanks. Set up a global rate limiter for all requests, similarly to what we had before. I think this should work.

**Review by amnn — APPROVED** [review]
:+1:

**Pull request merged**